cmd/pd-server/main.go

```diff
@@ -41,43 +42,88 @@ import (
-		syscall.SIGTERM,
-		syscall.SIGQUIT)
+	start()
+}
 
-	var sig os.Signal
-	go func() {
-		sig = <-sc
-		cancel()
-	}()
+// start main command
+func start() {
+	rootCmd := &cobra.Command{
+		Use:   "pd-server",
+		Short: "Placement Driver server",
+		Run:   createServerWrapper,
+	}
 
-	if err := svr.Run(); err != nil {
-		log.Fatal("run server failed", errs.ZapError(err))
+	rootCmd.Flags().BoolP("version", "V", false, "print version information and exit")
+	rootCmd.Flags().StringP("config", "", "", "config file")
+	rootCmd.Flags().BoolP("config-check", "", false, "check config file validity and exit")
+	rootCmd.Flags().StringP("name", "", "", "human-readable name for this pd member")
+	rootCmd.Flags().StringP("data-dir", "", "", "path to the data directory (default 'default.${name}')")
+	rootCmd.Flags().StringP("client-urls", "", "http://127.0.0.1:2379", "url for client traffic")
+	rootCmd.Flags().StringP("advertise-client-urls", "", "", "advertise url for client traffic (default '${client-urls}')")
+	rootCmd.Flags().StringP("peer-urls", "", "http://127.0.0.1:2379", "url for peer traffic")
+	rootCmd.Flags().StringP("advertise-peer-urls", "", "", "advertise url for peer traffic (default '${peer-urls}')")
+	rootCmd.Flags().StringP("initial-cluster", "", "", "initial cluster configuration for bootstrapping, e,g. pd=http://127.0.0.1:2380")
+	rootCmd.Flags().StringP("join", "", "", "join to an existing cluster (usage: cluster's '${advertise-client-urls}'")
+	rootCmd.Flags().StringP("metrics-addr", "", "", "prometheus pushgateway address, leaves it empty will disable prometheus push")
+	rootCmd.Flags().StringP("log-level", "L", "info", "log level: debug, info, warn, error, fatal (default 'info')")
+	rootCmd.Flags().StringP("log-file", "", "", "log file path")
+	rootCmd.Flags().StringP("cacert", "", "", "path of file that contains list of trusted TLS CAs")
+	rootCmd.Flags().StringP("cert", "", "", "path of file that contains X509 certificate in PEM format")
+	rootCmd.Flags().StringP("key", "", "", "path of file that contains X509 key in PEM format")
+	rootCmd.Flags().BoolP("force-new-cluster", "", false, "force to create a new one-member cluster")
+	rootCmd.AddCommand(NewServiceCommand())
+
+	rootCmd.SetOutput(os.Stdout)
+	if err := rootCmd.Execute(); err != nil {
+		rootCmd.Println(err)
+		os.Exit(1)
 	}
+}
 
-	<-ctx.Done()
-	log.Info("Got signal to exit", zap.String("signal", sig.String()))
+// NewServiceCommand returns the service command.
+func NewServiceCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "service <tso> [args]",
+		Short: "Run a service",
+	}
+	cmd.AddCommand(NewTSOServiceCommand())
+	return cmd
+}
 
-	svr.Close()
-	switch sig {
-	case syscall.SIGTERM:
-		exit(0)
-	default:
-		exit(1)
+// NewTSOServiceCommand returns the unsafe remove failed stores command.
+func NewTSOServiceCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "tso [args]",
+		Short: "run a tso service",
+		Run:   tso.CreateServerWrapper,
```

Comment:
so this is why we removed the return data types and have to duplicate server run() and exit code. LGTM